account-kit/smart-contracts/src/ma-v2/client/client.ts

```diff
@@ -64,7 +97,21 @@ export function createSMAV2AccountClient<
  */
 export async function createSMAV2AccountClient(
   config: CreateSMAV2AccountClientParams
-): Promise<SmartAccountClient> {
+): Promise<SmartAccountClient | AlchemySmartAccountClient> {
+  const { transport, chain } = config;
+  if (isAlchemyTransport(transport, chain)) {
+    const smaV2Account = await createSMAV2Account({
+      ...config,
+      transport,
+      chain,
+    });
```

Comment:
nit: we should be able to just pull this up and remove line 115 below since the account instance can be created the same way regardless of the client